### PR TITLE
[Merged by Bors] - Chore: Weaken hypotheses for proof that Noetherian is an extension property

### PR DESCRIPTION
**Pull request by Shamrock-Frost**
The middle term of a three term exact sequence with outer terms Noetherian is Noetherian. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Noetherian.lean [hunk 1]
```diff
@@ -119,6 +119,11 @@ theorem isNoetherian_of_surjective (f : M →ₗ[R] P) (hf : LinearMap.range f =
 
 variable {M}
 
+instance isNoetherian_quotient {R} [Ring R] {M} [AddCommGroup M] [Module R M]
```
Comment (erdOne) [inline Mathlib/RingTheory/Noetherian.lean:122]: Is there a reason behind the rename?

**Comment by erdOne** [comment]
Also please start the PR title with `feat:` or `chore:` etc. See https://leanprover-community.github.io/contribute/commit.html for details.

File: Mathlib/RingTheory/Noetherian.lean (hunk 1 above)
Comment (erdOne) [reply Mathlib/RingTheory/Noetherian.lean:122]: Sure. Can you add
```
@[deprecated] alias Submodule.Quotient.isNoetherian := isNoetherian_quotient -- deprecated on 2024-04-27
```
to the next line?

Comment (Shamrock-Frost) [reply Mathlib/RingTheory/Noetherian.lean:122]: Done.

**Comment by erdOne** [comment]
Kind reminder: this still fails CI

**Comment by Shamrock-Frost** [comment]
Sorry, I was headed into a movie and trying to get the PR fixed on my phone before it started (I did not succeed).

**Comment by erdOne** [comment]
maintainer merge

**Comment by erdOne** [comment]
No worries. Thanks for the contribution!

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**